Dialogue Addition

TikkuTokku opened the pull request:
Regression Dialogue Folder was added.

It has a manifest that uses Content Patcher.

The content.JSON tells Content Patcher to look in the Dialogue folder for changes.

When talking to Vincent or Jas for the first time, the player will get a question about whether or not they'd want ABDL lines for them. If they choose no, all ABDL changes for them will be replaced with normal dialogue. This includes reactions to the player using their pants.

Jodi has full dialogue for every day of the year.

Sam has full ABDL dialogue and full marriage ABDL dialogue.

Vincent has full dialogue for every day of the year.

Sebastion has one ABDL question for when I was learning how to do that.

Jas has the introduction question.

en.JSON was edited to have less humilation, full Jodi and Sam reactions, and Vincent which are avoided if the intro question was answered no.

zippity21 commented:
Alright, round 2 :)

For future reference, as long as you're making changes on the same branch, I don't believe you need a new PR. So you don't need to close this to resolve issues.

1) Can you update the manifest for your dialogue content to have the "contentpackedfor" field be Zippity21.Regression? This ensures that nobody tries to use it without this mod installed.

2) Jodi's Introduction still includes "This is dialogue!" Which I'm guessing was a test. So it's be good to remove it.

3) This is a logistical problem: I think I need to explain my situation better regarding the folder placement. You're right that putting the Regression folder and this content mod folder at the same level works, but it doesn't match how SMAPI packages the mod for release. SMAPI created a zip called "Regression vX.X.X", but in that folder is only a single other folder "Regression" which is where all the main mod's stuff is, then "Regression Dialogue" is underneath that. Due to that release structure, "Regression Dialogue" isn't detected unless you manually move it a directory up.


As long as you resolve 1 & 2, I'll perform the PR, but I think it's on me to figure out a resolution to 3. It sounds simple, bit it's quite the headache. I spent a couple days last weekend trying to figure out a solution. It'll probably be to disable SMAPI's auto packaging and just incorporate my own post-build action.

TikkuTokku commented:
You're right, I was able to just edit this pull request! That is very convenient.

1. The manifest needs to have content patcher in the "contentpacked" for field or it won't work. To achieve the same goal though, I added a dependency which includes Regression's unique ID. I gave it a quick test and the Regression Dialogue does not work without the Regression mod. 

2. I looked really hard to find that and couldn't. I did notice that I accidentally deleted the unique introduction I had for Jodi when I was cleaning things up, so I rewrote that. I also noticed a slight error on a Sam line where I missed a %. 

3. What if you had a folder called "Regression Mod" that contained both Regression and Regression dialogue. That way, SMAPI will create a zip called "Regression Mod vX.X.X" and in that folder is only a single other folder called "Regression Mod" which contains "Regression" and "Regression Dialogue" on the same level.

TikkuTokku commented:
Well, I'm poking around with seeing if my folder idea is actually viable, and I'm noticing a lot of code I already updated in this pull request. I think my inexperience with Github is causing some of the issues, like the Jody text you that is definitely not in the code right now.

zippity21 commented:
The issue is how SMAPI handles the post-build in the the visual studio project, so if you weren't creating a release through that flow, you wouldn't see the issue. I pushed a change to this branch that resolves this infrastructure issue. So everything looks good. I'll pull.

(Sorry for the long delay, its been quite the week)
